sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1149,75 +1149,60 @@ class FilterPushdownSuite extends PlanTest {
     comparePlans(Optimize.execute(originalQuery.analyze), correctAnswer)
   }
 
-  test("join condition pushdown: deterministic and non-deterministic") {
```

Comment:
This was added in https://github.com/apache/spark/pull/14815

It's not valid anymore, as we don't allow nondeterministic join condition now.